stream-transformation-tool-service/src/main/java/uk/gov/justice/tools/eventsourcing/transformation/service/EventStreamTransformationService.java

```diff
@@ -75,7 +75,7 @@ public UUID transformEventStream(final UUID originalStreamId, final int pass) {
 
             if (action.isTransform()) {
                 final Optional<UUID> newStreamId = eventTransformationStreamIdFilter.getEventTransformationStreamId(eventTransformations, jsonEnvelopeList);
-                if (newStreamId.isPresent()) {
+                if (newStreamId.isPresent() && !newStreamId.get().equals(originalStreamId)) {
```

Comment:
Can this check be created as a method or value and have a descriptive name?

Reply:
DONE